protected/humhub/modules/user/controllers/RegistrationController.php

```diff
@@ -91,29 +91,42 @@ public function actionIndex()
         } elseif (Yii::$app->session->has('authClient')) {
             $authClient = Yii::$app->session->get('authClient');
             $this->handleAuthClientRegistration($authClient, $registration);
-        } else {
-            Yii::warning('Registration failed: No token (query) or authclient (session) found!', 'user');
-            Yii::$app->session->setFlash('error', 'Registration failed.');
-            return $this->redirect(['/user/auth/login']);
         }
 
-        if ($registration->submitted('save') && $registration->validate() && $registration->register($authClient)) {
-            Yii::$app->session->remove('authClient');
+        if ($registration->submitted('save') && $registration->validate()) {
+            $existingUser = User::findIdentity($registration->getUser()->email);
```

Comment:
It's supposed to be a check to see if an existing user has a matching email then bypasses the registration process, if there is a better solution then I'm all for this.

Reply:
For me it looks like the `$existingUser` will be always `null` here because if for example:

- `$registration->getUser()->email` = `address@mail.com`
- `User::findIdentity('address@mail.com');` => `User::findOne(['id' => 'address@mail.com'])`

I think we should either pass `$registration->getUser()->id` instead of `email` or find user by email like `User::findOne(['email' => 'address@mail.com'])`.

Reply:
> For me it looks like the `$existingUser` will be always `null` here because if for example:
> 
> * `$registration->getUser()->email` = `address@mail.com`
> * `User::findIdentity('address@mail.com');` => `User::findOne(['id' => 'address@mail.com'])`
> 
> I think we should either pass `$registration->getUser()->id` instead of `email` or find user by email like `User::findOne(['email' => 'address@mail.com'])`.

Good point, I'll have to run some tests for this method.

Reply:
I'm currently not seeing any issues with either methods the process is the same when an OAuth client is used as well as normal registration.

Reply:
> I am wondering how user may be found by ID when email is passed as param where ID is expected.

`$registration->getUser()->email` and `$registration->getUser()->id` are properties that belong to the `User` model instance associated with the `Registration` model would be my guess?

Reply:
Yes, but `email` != `id`.

I see the code `User::findIdentity($registration->getUser()->email)` executes SQL query like `SELECT * FROM user WHERE id = {$registration->getUser()->email}`.

Why did you implement it as `User::findIdentity($registration->getUser()->id)`?
Or if you need to find a user by email then we should use another method to get a user by email like `User::findOne(['email' => $registration->getUser()->email])`.

Reply:
> Yes, but `email` != `id`.
> 
> I see the code `User::findIdentity($registration->getUser()->email)` executes SQL query like `SELECT * FROM user WHERE id = {$registration->getUser()->email}`.
> 
> Why did you implement it as `User::findIdentity($registration->getUser()->id)`? Or if you need to find a user by email then we should use another method to get a user by email like `User::findOne(['email' => $registration->getUser()->email])`.

I'm still not sure why both are working for me as I'm using the current develop branch as always to keep updated on latest changes, I'm sure it's not a cache issue and looking over the codebase would require too much time for me currently. 🤔

Reply:
It is strange why for you the code like `User::findOne(['id' => 'address@mail.com'])` can really find a user.
However if the method is declared as `public static function findIdentity($id)` with doc text `Finds an identity by the given ID.` then we should pass user id as param and not email.

Reply:
> It is strange why for you the code like `User::findOne(['id' => 'address@mail.com'])` can really find a user. However if the method is declared as `public static function findIdentity($id)` with doc text `Finds an identity by the given ID.` then we should pass user id as param and not email.

I'm not sure that going by `id` would be the best approach here as `ids` are unique to the user and can be different on other platforms that the auth client is used for, using email would be better because it's something that both accounts should have in common and are more commonly allowed. The same can be said if we were to check against `username` instead of `email`, what's used on one platform can be completely different on another.

Reply:
If you need to find a user by email then you should use a code like `User::findOne(['email' => $registration->getUser()->email])` instead of the current `User::findIdentity($registration->getUser()->email)` because this method searches users by id.

Reply:
Updated here https://github.com/humhub/humhub/commit/4c7f67f0277c190246bd738b55765c8f6904be99

Reply:
Thank you, now the code looks correctly.